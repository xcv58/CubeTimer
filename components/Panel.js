import React from 'react'
import { inject, observer } from 'mobx-react'
import Button from '@material-ui/core/Button'

@inject('store')
@inject('recordsStore')
@observer
export default class Panel extends React.Component {
  clear = () => {
    const { recordsStore, store } = this.props
    recordsStore.clear()
    store.clear()
  }

  render () {
    const { recordsStore: { records }, store: { running, lapse } } = this.props
    if (records.length < 1 && (running || lapse === 0)) {
      return null
    }
    return (
      <div className='timebar' style={{
        display: 'flex',
        justifyContent: 'center'
      }}>
        <Button onClick={this.clear}
          disabled={running}
          style={{ width: '100%' }}>
          Clear
        </Button>
      </div>
    )
  }
}
