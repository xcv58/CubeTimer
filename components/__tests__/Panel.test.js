/* global it, expect, describe */
import React from 'react'
import Panel from '../Panel'
import { spy } from 'sinon'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Button from '@material-ui/core/Button'

describe('Panel', () => {
  it(
    'render null if records is empty and stopwatch is running or lapse is 0',
    () => {
      const recordsStore = { records: [] }
      const store = { running: true, lapse: 0 }
      let el = mount(<Panel {...{ recordsStore, store }} />)
      expect(el).toBeTruthy()
      expect(el.text()).toBe(null)
      el = mount(<Panel {...{
        recordsStore,
        store: { running: false, lapse: 0 }
      }} />)
      expect(el).toBeTruthy()
      expect(el.text()).toBe(null)
      el = mount(<Panel {...{
        recordsStore,
        store: { running: true, lapse: 100 }
      }} />)
      expect(el).toBeTruthy()
      expect(el.text()).toBe(null)

      const tree = renderer.create(
        <Panel {...{ recordsStore, store }} />
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })

  it('render clear and call clear onClick', () => {
    const recordsStore = { records: [ 42 ], clear: spy() }
    const store = { clear: spy() }
    const props = { store, recordsStore }
    const expectedText = 'Clear'
    const el = mount(<Panel {...props} />)
    expect(el.find(Button).is({ children: expectedText })).toBe(true)
    expect(el.find(Button).exists()).toBe(true)
    expect(store.clear.callCount).toBe(0)
    expect(recordsStore.clear.callCount).toBe(0)
    el.find(Button).props().onClick()
    expect(store.clear.callCount).toBe(1)
    expect(recordsStore.clear.callCount).toBe(1)
    el.find(Button).props().onClick()
    expect(store.clear.callCount).toBe(2)
    expect(recordsStore.clear.callCount).toBe(2)

    const tree = renderer.create(
      <Panel {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
