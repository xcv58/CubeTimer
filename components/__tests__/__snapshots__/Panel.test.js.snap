// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel render clear and call clear onClick 1`] = `
<div
  className="timebar"
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <button
    className="MuiButtonBase-root-23 MuiButton-root-1 MuiButton-textSecondary-4 MuiButton-flat-5 MuiButton-flatSecondary-7"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      Clear
    </span>
    <span
      className="MuiTouchRipple-root-26"
    />
  </button>
</div>
`;

exports[`Panel render null if records is empty and stopwatch is running or lapse is 0 1`] = `null`;
